src/main/java/org/broadinstitute/consent/http/service/UserService.java

```diff
@@ -91,8 +98,9 @@ public User updateUserFieldsById(UserUpdateFields userUpdateFields, Integer user
                 userPropertyDAO.deletePropertiesByUserAndKey(userProps);
                 userPropertyDAO.insertAll(userProps);
             }
-            // Handle Roles.
-            if (Objects.nonNull(userUpdateFields.getUserRoleIds())) {
+
+            // Handle Roles; must be admin to update roles.
+            if (user.hasUserRole(UserRoles.ADMIN) && Objects.nonNull(userUpdateFields.getUserRoleIds())) {
```

Comment:
Same comment here, I would prefer not to check for permissions here. If there were business logic, i.e. admins can update roles XYZ, others can update roles EFG, then yes, that makes sense here. I think there are two easy options here. One is to break out the update method into two, one for handling all values in `UserUpdateFields`, and then another separate one for not handling roles. A second option is in the resource class, make sure that there are no roles in the `UserUpdateFields` object. That can be enforced in the resource method and throwing a BadRequestException if a user tries to update their own roles in this manner.

Reply:
makes perfect sense! i separated out the logic so that all auth logic is at the resource level, should address both comments. i appreciate the thorough help here!